alf/algorithms/dsac_algorithm.py

```diff
@@ -266,39 +263,22 @@ def _compute_critics(self,
 
     def _critic_train_step(self, inputs: TimeStep, state: SacCriticState,
                            rollout_info: SacInfo, action, tau_info: TauInfo):
-        bs = self._mini_batch_size
         # Calculate the critics and value for both the current observation
         # and next observation
-        # [B * (T-1), n_quantiles]
-        critic_observation = nest.map_structure(lambda tensor: tensor[:-bs],
-                                                inputs.observation)
-        critic_action = nest.map_structure(lambda tensor: tensor[:-bs],
-                                           rollout_info.action)
+        # [B * T, n_quantiles] or [B, n_quantiles]
         critics, critics_state, _ = self._compute_critics(
             self._critic_networks,
-            critic_observation,
-            critic_action,
-            tau_info.tau_hat[:-bs],
+            inputs.observation,
+            rollout_info.action,
+            tau_info.tau_hat,
```

Comment:
what's the reason for this change?

Reply:
Previously, both ``inputs.observation`` and ``rollout_info.action`` are chunked to extract the first [B * (T-1), ...] tensors to forward the critics, instead of forwarding the full tensors of shape [B * T, ...] as done by SAC and other alf algoirthms, to save some computation time, since they will be chunked either way during TD loss computation. However, this will not work with the case when ``temporally_independent_train_step=False``. So I just changed back to the default case.